### PR TITLE
python312Package.ssort: init at 0.13.0

### PR DESCRIPTION
**Pull request by Tochiaha**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [x] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Tochiaha** [comment]
@fabaff

File: pkgs/development/python-modules/ssort/default.nix
Comment (Sigmanificient) [inline pkgs/development/python-modules/ssort/default.nix:1]:
```suggestion
{
  lib,
  fetchFromGitHub,
  buildPythonPackage,
  setuptools,
  pathspec,
}:
```

File: pkgs/development/python-modules/ssort/default.nix
Comment (Sigmanificient) [inline pkgs/development/python-modules/ssort/default.nix:22]:
```suggestion
  pythonImportsCheck = [ "ssort" ];
  nativeCheckInputs = [
    pytestCheckHook
    pyyaml
  ];

  preCheck = ''
    substituteInPlace tests/test_executable.py \
      --replace-fail '["ssort"]' "['${placeholder "out"}/bin/ssort']" \
      --replace-fail '["ssort",' "['${placeholder "out"}/bin/ssort',"
  '';
```

File: pkgs/development/python-modules/ssort/default.nix
Comment (Sigmanificient) [inline pkgs/development/python-modules/ssort/default.nix:30]: Iirc you dont need to set `platforms` for python modules, may a other maintainer confirm this
```suggestion
```

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

**Comment by Tochiaha** [comment]
@superherointj

File: pkgs/development/python-modules/ssort/default.nix
Comment (natsukium) [inline pkgs/development/python-modules/ssort/default.nix:38]: Can we use `export PATH=$out/bin:$PATH` instead?

Comment (Tochiaha) [reply pkgs/development/python-modules/ssort/default.nix:38]: but this is ok too.

Comment (superherointj) [reply pkgs/development/python-modules/ssort/default.nix:38]: > but this is ok too.

We need **consensus** to move forward. Either you wait for natsukium to answer. Or you can try his solution.
Goal is mostly to reach consensus. And yes, there are many different ways to do things. And there isn't one right.
I can only tell you what I would do. I would try his solution. And see if it works fine. But then, you do you. You could wait him. I know this is something basic and simple. And not important, yes... I know. Your solution works too.
I'm just trying to offer a shortcut here, while he hasn't answered you. Maybe it can speed things a bit. So you get your PR merged sooner rather than later.
Hope it makes sense.

Comment (Tochiaha) [reply pkgs/development/python-modules/ssort/default.nix]: i had natsukium solution in preCheck if that is not ok, i can still make the changes. Once again thank you for your feedback.

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 324722` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.ssort</li>
    <li>python311Packages.ssort.dist</li>
    <li>python312Packages.ssort</li>
    <li>python312Packages.ssort.dist</li>
  </ul>
</details>

**Review by superherointj — APPROVED** [review]

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 324722` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.ssort</li>
    <li>python311Packages.ssort.dist</li>
    <li>python312Packages.ssort</li>
    <li>python312Packages.ssort.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/ssort/default.nix
Comment (natsukium) [reply pkgs/development/python-modules/ssort/default.nix]: Sorry for the lack of explanation.
If patched, the patch needs to be updated every time the source is updated.
Adding environment variables is a more common solution and a more stable way.

**Pull request merged**
